posixlib/src/main/scala/scala/scalanative/posix/sys/wait.scala

```diff
@@ -0,0 +1,43 @@
+package scala.scalanative.posix.sys
+
+import scala.scalanative.unsafe._
+import scala.scalanative.posix.sys.types.pid_t
+
+@extern
+object wait {
+  @name("scala_native_wait_WCONTINUED")
+  val WCONTINUED: CInt = extern
+
+  @name("scala_native_wait_WNOHANG")
+  val WNOHANG: CInt = extern
+
+  @name("scala_native_wait_WUNTRACED")
+  val WUNTRACED: CInt = extern
+
+  @name("scala_native_wait_WIFEXITED")
+  def WIFEXITED(statVal: CInt): CInt = extern
+
+  @name("scala_native_wait_WEXITSTATUS")
+  def WEXITSTATUS(statVal: CInt): CInt = extern
+
+  @name("scala_native_wait_WIFSIGNALED")
+  def WIFSIGNALED(statVal: CInt): CInt = extern
+
+  @name("scala_native_wait_WTERMSIG")
+  def WTERMSIG(statVal: CInt): CInt = extern
+
+  @name("scala_native_wait_WIFSTOPPED")
+  def WIFSTOPPED(statVal: CInt): CInt = extern
+
+  @name("scala_native_wait_WSTOPSIG")
+  def WSTOPSIG(statVal: CInt): CInt = extern
+
+  @name("scala_native_wait_WIFCONTINUED")
+  def WIFCONTINUED(statVal: CInt): CInt = extern
+
+  //renamed to avoid shadowing by with Object.wait
```

Comment:
I will have to trace through the shadowing issue when I have more time
and/or smarts.  I suspect that the "shadowing" is a non-issue or
expected behavior, but do not want to say so without evidence or 
careful thought.  

The "@name" is not wrong, just a piece of complexity.

For now I think the "by with" can be deleted or reworded as something like
 "To avoid wait.wait() shadowing Object.wait()"

Reply:
Ah, the problem is that wait.wait() will be shadowed by Object.wait() in a normal import.

Reply:
My eyes may be blurry before my morning coffee, but I see no zero arg
`wait.wait()` in the 2018 spec.   I see a one arg, and two three arg prototypes.

`man wait`, which documents the glibc variant of clib, has a zero argument waitpid().

By my reading & expectation, the one argument wait.wait `pid_t  wait(int *);`
and the one argument Object.wait `void | wait(long timeout)`
 should be distinct and distinguishable because the corresponding Scala 
signatures (should) differ.

Am I missing something or tracing incorrectly?  


Are you seeing different behavior: i.e. that Object.wait(timeout) is being
called when `sys/wait.scala` is in scope when the "@name" under discussion is removed?  If yes, I will clone this PR and dive deeper.  

This discussion is exactly the benefit of SN providing the `posixlib`
definitions and "glue" layer.  Someone took the time to get the 
(big!) and tiny pain points resolved instead of relying upon a thousand
developers to write their own and get it wrong some fraction of the time.

Reply:
Switching to wait results in this error message in `WaitTest`

```scala
[error] /home/mhammons/Documents/Projects/scala-native/unit-tests/src/test/scala/scala/scalanative/posix/sys/WaitTest.scala:33:27: type mismatch;
[error]  found   : scala.scalanative.unsafe.Ptr[scalanative.unsafe.CInt]
[error]     (which expands to)  scala.scalanative.unsafe.Ptr[Int]
[error]  required: Long
[error]       val childPid = wait(statLoc)
```

Reply:
I believe that the compiler is doing its job and telling us that `wait.wait(statloc)` is not in scope, when
we think it is.  In other words, the same mis-implementation as needing to do `errno.errno`.  I am copying
over the code of this PR to try out by suspicion that making the object initial caps solves this issue.
More news as it develops.

Reply:
I cloned the originating repository and spend a long evening chasing this.  Looks to me
like it is 
1) another instance of the long running `errno.errno` trauma
2) A compiler bug or at least an undocumented inter-op quirk.

I must stop for the night, but will try some more to understand this
and see if I can create a smaller reproducer.

Looks very much like symbol resolution is broken (for @extern classes?)
and probably very early on.  I need to try a Scala JVM example to make
sure my understanding of how the signatures are supposed to be matched 
is correct.

Reply:
ok. can we clear this issue by renaming as waitpid until this symbol resolution bug is fixed? (I can create an issue to rename the function that we can mark as blocked by this new bug).

what do you think?

Reply:
I am good with creating an Issue, saying that the underlying signature resolution handling
is under investigation, and then a comment in the wait.scala code giving that Issue number.
I can not speak for them, but I suspect that would not pass Scala Center review (which is
what gates a merge).

Can I propose an alternate approach as a basis for discussion/iteration?   To reduce 
the amount of complexity, lore, and hanging chads (incomplete or undecipherable items)
to a minimum, what do you think about using the proper `wait(statLoc)` declaration
with a comment directly above it saying something like "// Must be used as 'wait.wait(statLoc) or else compiler uses Object.wait()'.    

Yes, the lets the requirement escape into users, where your solution isolates the "WT?" to developers
reading the file.  It does reduce the "WT?"  like why was the name `waitPid` chosen and not `waitX()` or
`wait__()`. Both of those also fail in my study environment.

My proposal may not pass Scala Center review either.  They may want this PR to be paused until
a proper fix is discovered.  Let's see what sorts out.        

Can your private work carry on with private copies containing your fix or do you need a -SNAPSHOT
fix?  Understood, getting things merged in a timely manner is good all around.

To help this PR along, I am actively studying the underlying `errno.errno`/`wait.wait` problem.
with finding signatures (in extern objects?).